cfg/cfg.go

```diff
@@ -1,266 +1,252 @@
-	}
-
-	tv, err := time.Parse(time.UnixDate, value)
-	if err != nil {
-		panic(fmt.Sprintf("Key %q value is not a Time", key))
-	}
-
-	return tv
-}
-
-// Bool returns the bool balue of a given key as a bool, else it will return an
-// error, if the key was not found or the value can't be convered to a bool.
-func Bool(key string) (bool, error) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	value, found := c.m[key]
-	if !found {
-		return false, fmt.Errorf("Unknown Key %s !", key)
-	}
-
-	if value == "on" || value == "yes" {
-		value = "true"
-	} else if value == "off" || value == "no" {
-		value = "false"
-	}
-
-	val, err := strconv.ParseBool(value)
-	if err != nil {
-		return false, err
-	}
-
-	return val, nil
-}
-
-// MustBool returns the bool balue of a given key as a bool, else it will panic
-// if the key was not found or the value can't be convered to a bool.
-func MustBool(key string) bool {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	value, found := c.m[key]
-	if !found {
-		panic(fmt.Sprintf("Unknown Key %s !", key))
-	}
-
-	if value == "on" || value == "yes" {
-		value = "true"
-	} else if value == "off" || value == "no" {
-		value = "false"
-	}
-
-	val, err := strconv.ParseBool(value)
-	if err != nil {
-		return false
-	}
-
-	return val
-}
-
-// URL returns the value of the giving key as a URL, else it will return an
-// error, if the key was not found or the value can't be convered to a URL.
-func URL(key string) (*url.URL, error) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	value, found := c.m[key]
-	if !found {
-		return nil, fmt.Errorf("Unknown Key %s !", key)
-	}
-
-	u, err := url.Parse(value)
-	if err != nil {
-		return u, err
-	}
-
-	return u, nil
-}
-
-// MustURL returns the value of the giving key as a URL, else it will panic
-// if the key was not found or the value can't be convered to a URL.
-func MustURL(key string) *url.URL {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	value, found := c.m[key]
-	if !found {
-		panic(fmt.Sprintf("Unknown Key %s !", key))
-	}
-
-	u, err := url.Parse(value)
-	if err != nil {
-		panic(fmt.Sprintf("Key %q value is not a URL", key))
-	}
-
-	return u
-}
+package cfg
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+// c represents the configuration store, with a map to store the loaded keys
+// from the environment.
+var c struct {
+	m  map[string]string
+	mu sync.RWMutex
+}
+
+//==============================================================================
+
+// Provider is implemented by the user to provide the configuration as a map
+type Provider interface {
+	Provide() (map[string]string, error)
+}
+
+//==============================================================================
+
+// Init is to be called only once, to load up the giving namespace if found,
+// in the environment variables. All keys will be made lowercase.
+func Init(p Provider) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	// get the provided configuration
```

Comment:
Comments must use proper sentence structure. Check the rest of the code.

// Get the provided configuration.